### PR TITLE
Use custom validator for pulp url

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
Not sure how i didn't catch this before...

**Comment by jlsherrill** [comment]
added a test to go through the full plugin initialization that would have caught the error

**Comment by jlsherrill** [comment]
[test smart_proxy_pulp]

**Comment by jlsherrill** [comment]
@ekohl if you get a few moments :)

File: test/pulp3_plugin_test.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+class PulpApiTest < Test::Unit::TestCase
+  def test_loaded
+    ::Proxy::DefaultModuleLoader.any_instance.stubs(:load_configuration_file).returns(:enabled => true,
+                                                                                      :pulp_url => 'http://foo/pulp/')
+
+    Proxy::PluginInitializer.new(::Proxy::Plugins.instance).initialize_plugins
+
+    assert_equal :running, ::Proxy::Plugins.instance.find{|plugin| plugin[:name] == :pulp3}[:state]
+  end
+end
```
Comment (ekohl) [inline test/pulp3_plugin_test.rb]: Nit: I'm a fan of line endings here

File: test/pulp3_plugin_test.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require 'smart_proxy_pulp_plugin/pulp3_plugin'
+
+class PulpApiTest < Test::Unit::TestCase
+  def test_loaded
+    ::Proxy::DefaultModuleLoader.any_instance.stubs(:load_configuration_file).returns(:enabled => true,
```
Comment (ekohl) [inline test/pulp3_plugin_test.rb]: Could add `.with('pulp3.yml')` to be more specific.

**Comment by jlsherrill** [comment]
I do like the integration test better, updated

File: test/pulp3_plugin_integration_test.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  def app
+    Proxy::PluginInitializer.new(Proxy::Plugins.instance).initialize_plugins
+    Proxy::RootV2Api.new
+  end
+
+  def test_features
+    Proxy::DefaultModuleLoader.any_instance.expects(:load_configuration_file).with('pulp.yml').returns(enabled: true, pulp_url: 'http://foo.com/foo')
+    Proxy::DefaultModuleLoader.any_instance.expects(:load_configuration_file).with('pulpnode.yml').returns(enabled: true, pulp_url: 'http://foo.com/foo')
+    Proxy::DefaultModuleLoader.any_instance.expects(:load_configuration_file).with('pulp3.yml').returns(enabled: true, pulp_url: 'http://foo.com/foo')
```
Comment (ekohl) [inline test/pulp3_plugin_integration_test.rb]: Could you change this to `http://pulp3.example.com/foo` (and use example.com in other tests as well)? That guarantees we're reading the correct settings file as well.

Comment (jlsherrill) [reply test/pulp3_plugin_integration_test.rb]: Good idea!.  done!

**Comment by jlsherrill** [comment]
@ehelms @ekohl can i get a merge and a gem bump :)

**Comment by ekohl** [comment]
I can actually (ab)use my admin permissions for the merge part.

**Pull request merged**